chemprop/uncertainty/uncertainty_calibrator.py

```diff
@@ -733,6 +730,281 @@ def nll(self, preds: List[List[float]], unc: List[List[float]], targets: List[Li
+        uncal_preds = np.array(
+            self.calibration_predictor.get_uncal_preds()
+        )  # shape(data, tasks, num_classes)
+        targets = np.array(self.calibration_data.targets(), dtype=bool)  # shape(data, tasks)
+        self.num_data, self.num_tasks = targets.shape
+
+        has_zeros = np.any(targets == 0, axis=1)
+        inds_zeros = targets[has_zeros] == 0
+        scores_in = self.nonconformity_scores(uncal_preds[has_zeros])
+        masked_scores_in = scores_in * inds_zeros + np.nan_to_num(
+            np.inf * (1 - inds_zeros).astype(float)
+        )
+        calibration_scores_in = np.min(masked_scores_in, axis=1)
+
+        has_ones = np.any(targets == 1, axis=1)
+        inds_ones = targets[has_ones] == 1
+        scores_out = self.nonconformity_scores(uncal_preds[has_ones])
+        masked_scores_out = scores_out * inds_ones + np.nan_to_num(
+            -np.inf * (1 - inds_ones).astype(float)
+        )
+        calibration_scores_out = np.max(masked_scores_out, axis=1)
+
+        self.tout = np.quantile(calibration_scores_out, 1 - self.conformal_alpha / 2)
+        self.tin = np.quantile(calibration_scores_in, self.conformal_alpha / 2)
+
+    def apply_calibration(self, uncal_predictor: UncertaintyPredictor):
+        uncal_preds = np.array(uncal_predictor.get_uncal_preds())  # shape(data, task)
+        scores = self.nonconformity_scores(uncal_preds)
+
+        cal_preds_in = (scores <= self.tin).astype(int)
+        cal_preds_out = (scores <= self.tout).astype(int)
+        cal_preds = np.concatenate((cal_preds_in, cal_preds_out), axis=1)
+
+        return uncal_preds.tolist(), cal_preds.tolist()
+
+
+class ConformalQuantileRegressionCalibrator(UncertaintyCalibrator):
+    """
+    Conformal Calibrator for regression datasets. Outputs interval of variable size, centered around
+    quantile outputs of model, for each datapoint. Intervals should cover 1-alpha proportion of datapoints.
+    As discussed in https://arxiv.org/abs/2107.07511.
+    """
+
+    @property
+    def label(self):
+        return "conformal_quantile_regression"
+
+    def raise_argument_errors(self):
+        super().raise_argument_errors()
+        if self.dataset_type != "regression":
+            raise ValueError(
+                "Conformal Quantile Regression is only implemented for regression dataset types."
+            )
+
+    def nll(
+        self,
+        preds: List[List[float]],
+        unc: List[List[float]],
+        targets: List[List[float]],
+        mask: List[List[bool]],
+    ):
+        raise NotImplementedError
+
+    @staticmethod
+    def get_preds(predictor: UncertaintyPredictor):
+        return np.array(predictor.get_uncal_preds())
+
+    def calibrate(self):
+        uncal_preds = self.get_preds(self.calibration_predictor)  # shape(data, 2 * tasks)
+        targets = np.array(self.calibration_data.targets(), dtype=float)  # shape(data, tasks)
+        targets = np.nan_to_num(targets)
```

Comment:
Instead of zeroing out missing targets, use a task masking approach like is used in other calibration methods. Missing values in training/calibration/evaluation data are very common and we need to be able to accommodate them without creating any false signal in the values. And having some placeholder zeros in here is something we should avoid.